src/sentry/llm/providers/base.py

```diff
@@ -0,0 +1,46 @@
+from typing import Any
+
+from sentry.llm.exceptions import InvalidModelError, InvalidProviderError
+from sentry.utils.services import Service
+
+
+class LlmModelBase(Service):
+    def __init__(self, **options: Any) -> None:
+        self.provider_config = options
+
+    def complete_prompt(
+        self,
+        *,
+        usecase_config: dict[str, Any],
```

Comment:
Is `usecase_config` going to be portable across different backends? Or will they be tied to the llm provider?

Reply:
kind of complicated --

the usecase config would be tied to the LLM provider, so it would look like:
```
"spam_detection": {
    "provider": "openai",
    "options": {
        "model": "chat-gpt-3.5-turbo",
    },
}

"ai_autofix": {
    "provider": "openai",
    "options": {
        "model": "chat-gpt-4-turbo",
    },
}
```

i thought about having a mapping of like "small model/"medium model"/"large model" that then maps to the LLM providers specific models, so that the usecaseconfigs are more portable, i'm not sure the added indirection there is worth it.